pkg/distsql/context/context.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import (
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/sessionctx/variable"
+)
+
+// DistSQLContext gives the interface
+type DistSQLContext interface {
+	// GetSessionVars gets the session variables.
+	GetSessionVars() *variable.SessionVars
+	// GetClient gets a kv.Client.
+	GetClient() kv.Client
```

Comment:
It's possible to remove `GetClient()` from the session context. It's only used in executor and distsql, but I'm not sure whether it's appropriate (or whether it's helpful). Therefore, I kept it here.